packages/web3-react/src/context/web3.tsx

```diff
@@ -0,0 +1,87 @@
+import invariant from 'tiny-invariant';
+import {
+  Web3Provider,
+  ExternalProvider,
+  JsonRpcFetchFunc,
+} from '@ethersproject/providers';
+import { CHAINS } from '@lido-sdk/constants';
+import { getStaticRpcBatchProvider } from '@lido-sdk/providers';
+import { ProviderSDK as ProviderSDKBase } from '@lido-sdk/react';
+import { Web3ReactProvider, useWeb3React } from '@web3-react/core';
+import { memo, FC } from 'react';
+import { SWRConfiguration } from 'swr';
+import { POLLING_INTERVAL } from '../constants';
+import ProviderConnectors, { ConnectorsContextProps } from './connectors';
+
+export interface ProviderWeb3Props extends ConnectorsContextProps {
+  defaultChainId: CHAINS;
+  supportedChainIds: CHAINS[];
```

Comment:
It should not be restricted to CHAINS, otherwise we are heavily limiting use cases

Reply:
It's okay for now